core/store/src/trie/mem/nibbles_utils.rs

```diff
@@ -0,0 +1,34 @@
+pub(crate) fn hex_to_nibbles(hex: &str) -> Vec<u8> {
+    if hex == "_" {
+        return vec![];
+    }
+    assert!(hex.len() % 2 == 0);
+    hex::decode(hex).unwrap()
```

Comment:
Is that right? Hex decodes to bytes but nibbles are half-bytes. I do realize you only moved it and it's probably ok so I'm mostly curious where is the gap in my understanding.

Reply:
I think the Robin's assumption was that we can only write actual bytes and we can't write key with odd number of nibbles. Maybe trie internals shouldn't be aware of that; on the other hand, vectors of even lengths cover all cases well.

Reply:
I was expecting the bytes to be further split down into nibbles, so something like this:
```
hex::decode(hex).unwrap().iter().map(|byte| vec![byte & 0xf0 >> 4, byte & 0x0f]).collect_vec().concat()
```

Reply:
No, each string symbol represents a readable nibble (0..9, a..f). I'll add comment